confignode/src/main/java/org/apache/iotdb/confignode/client/AsyncDataNodeClientPool.java

```diff
@@ -58,13 +57,106 @@ public class AsyncDataNodeClientPool {
 
   private final IClientManager<TEndPoint, AsyncDataNodeInternalServiceClient> clientManager;
 
+  private final int retryNum = 6;
+
   private AsyncDataNodeClientPool() {
     clientManager =
         new IClientManager.Factory<TEndPoint, AsyncDataNodeInternalServiceClient>()
             .createClientManager(
                 new ConfigNodeClientPoolFactory.AsyncDataNodeInternalServiceClientPoolFactory());
   }
 
+  /**
+   * When the request fails, reconnect the DataNode that failed the request
+   *
+   * @param req request
+   * @param handlerMap Map<index, Handler>
+   * @param dataNodeLocations ConcurrentHashMap<index, TDataNodeLocation> The specific DataNodes
+   */
+  public void sendAsyncRequestToDataNodeWithRetry(
+      Object req,
+      Object handlerMap,
+      ConcurrentHashMap<Integer, TDataNodeLocation> dataNodeLocations) {
+    CountDownLatch countDownLatch = new CountDownLatch(dataNodeLocations.size());
+    if (dataNodeLocations.isEmpty()) {
+      return;
+    }
+    for (int retry = 0; retry < retryNum; retry++) {
+      AbstractRetryHandler handler = null;
+      for (Map.Entry<Integer, TDataNodeLocation> entry : dataNodeLocations.entrySet()) {
+        handler = ((Map<Integer, AbstractRetryHandler>) handlerMap).get(entry.getKey());
+        // If it is not the first request, then prove that this operation is a retry.
+        // The count of countDownLatch needs to be updated
+        if (retry != 0) {
+          handler.setCountDownLatch(countDownLatch);
+        }
+        // send request
+        sendAsyncRequestToDataNode(entry.getValue(), req, handler, retry);
+      }
+      try {
+        handler.getCountDownLatch().await();
```

Comment:
```suggestion
        countDownLatch.await();
```
Since we generate a new CountDownLatch in each retry loop.

Reply:
fixed~